dataikuapi/govern/artifact_search_handler.py

```diff
@@ -0,0 +1,203 @@
+        "name", "workflow", "field".
+        :param str sort_direction: the direction on which the results will be sorted. The sort direction must be either
+        "ASC" or "DESC".
+        :return: the created :class:`govern.models.Blueprint`
+        :returns The query object that will perform the search.
+        :rtype: :class:`dataikuapi.govern.artifact_search_handler.GovernArtifactSearchRequest`
+        """
+
+        if ids_list is None:
+            ids_list = []
+
+        if artifact_search_source_type not in ["all", "blueprints", "blueprintVersions", "artifacts"]:
+            raise ValueError("""Parameter artifact_search_source must be chosen from "all", "blueprints", 
+            "blueprintVersions", or "artifacts".""")
+
+        artifact_search_source = {"type": artifact_search_source_type}
+        if artifact_search_source_type == "blueprints":
+            artifact_search_source["blueprintIds"] = ids_list
+        if artifact_search_source_type == "blueprintVersions":
+            artifact_search_source["blueprintVersionIds"] = ids_list
+        if artifact_search_source_type == "artifacts":
+            artifact_search_source["artifactIds"] = ids_list
+
+        if artifact_filters_list is None:
+            artifact_filters_list = []
+        query = {"artifactFilters": [artifact_filter.build() for artifact_filter in artifact_filters_list]}
+        if sort_direction is not None or sort_column is not None:
+            query["artifactSearchSort"] = {"direction": sort_direction, "column": {"type": sort_column}}
+
+        return GovernArtifactSearchQuery(artifact_search_source, query, )
+
+    def build_request(self, query):
+        """
+        Create a new artifact search request and return the object that will be used to launch the request
+
+        :param :class:`dataikuapi.govern.artifact_search_handler.GovernArtifactSearchQuery` query: The query that will
+        be addressed during the search.
+        :returns The created artifact search request object
+        :rtype: :class:`dataikuapi.govern.artifact_search_handler.GovernArtifactSearchRequest`
+        """
+
+        return GovernArtifactSearchRequest(self.client, query)
+
+
+class GovernArtifactSearchQuery(object):
+    """
+    An object which represents a query
+    """
+
+    def __init__(self, artifact_search_source, query):
+        self.artifact_search_source = artifact_search_source
+        self.query = query
+
+    def build_query(self):
+        return {"artifactSearchSource": self.artifact_search_source,
+                "query": self.query}
+
+
+class GovernArtifactSearchRequest(object):
+    """
+    A search request object.
+    Do not create this directly, use :meth:`dataikuapi.govern.artifact_search_handler.build_query()` and then run the
+    query using :meth:`dataikuapi.govern.artifact_search_handler.GovernArtifactSearchRequest.perform_search()`
+    """
+
+    def __init__(self, client, query):
```

Comment:
maybe this method can be called inside perform_search instead of the user needed to call it. and the query class would be passed directly to the GovernArtifactSearchRequest ? so query = instance of class GovernArtifactSearchQuery